Integrations/integration-AzureCompute.yml

```diff
@@ -0,0 +1,1085 @@
+
+
+    '''MAIN FUNCTIONS / API CALLS'''
+
+    #<-------- Resource Groups -------->#
+
+    def list_resource_groups():
+        endpoint_url = 'https://management.azure.com/subscriptions/' + SUBSCRIPTION_ID
+        endpoint_url += '/resourcegroups?api-version=2018-05-01'
+        update_access_token()
+        response = http_request('GET', full_url=endpoint_url, codes={200})
+        return response
+
+
+    def list_resource_groups_command():
+        """
+        List all Resource Groups belonging to your Azure subscription
+
+        returns:
+            Resource-Group Objects
+        """
+        response = list_resource_groups()
+        # Retrieve relevant properties to return to context
+        value = response.get('value')
+        resource_groups = []
+        for resource_group in value:
+            rg = {
+                'Name': resource_group.get('name'),
+                'ID': resource_group.get('id'),
+                'Location': resource_group.get('location'),
+                'ProvisioningState': resource_group.get('properties').get('provisioningState')
+            }
+            resource_groups.append(rg)
+
+        title = "List of Resource Groups"
+        human_readable = tableToMarkdown(title, resource_groups)
+        demisto.results({
+            'Type': entryTypes['note'],
+            'Contents': response,
+            'ContentsFormat': formats['text'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': human_readable,
+            'EntryContext': {'Azure.ResourceGroup(val.Name && val.Name === obj.Name)': resource_groups}
+        })
+
+
+    #<-------- Virtual Machines -------->#
+
+    def list_vms(resource_group):
+        # Construct endpoint URI suffix
+        url_endpoint = resource_group + '/providers/Microsoft.Compute/virtualMachines?api-version=' + API_VERSION
+        # Call API
+        update_access_token()
+        response = http_request('GET', url_endpoint, codes={200})
+        return response
+
+
+    def list_vms_command():
+        """
+        List the VM instances in the specified Resource Group
+
+        demisto parameter: (string) resource_group
+            Resource Group of the VMs
+
+        returns:
+            Virtual Machine Objects
+        """
+        resource_group = demisto.args().get('resource_group')
+        response = list_vms(resource_group)
+
+        num_of_vms = len(response.get('value'))
+
+        vms = []
+        for i in range(0, num_of_vms):
+            value = response.get('value')[i]
+            vm_name = value.get('name')
+            location = value.get('location')
+            properties = value.get('properties')
+            provisioning_state = properties.get('provisioningState')
+            os_disk = properties.get('storageProfile').get('osDisk')
+            datadisk = os_disk.get('diskSizeGB', 'NA')
+            vm_id = properties.get('vmId')
+            os_type = os_disk.get('osType')
+            vm = {
```

Comment:
not sure i understand why you're not inserting the values directly to the dict

Reply:
I thought it improved code readability but I wasn't entirely consistent across all the commands. If there was only 2 or 3 properties to fetch that were going to be returned to the context then I directly inserted the values into the dict - if it was a lot, I separated it.